### PR TITLE
`better-regexp`: bump `regexp-tree` dependency

### LINKED ISSUE #994: Broken autofix with `better-regex` and multiple optional spaces
The following throws a ` /\s?\s?/ can be optimized to /\s?{2}/       unicorn/better-regex`

```js
const regex = /\s?\s?/;
```

which auto-fixes to:

```js
const regex = /\s?{2}/;
```

This isn't valid though, and throws a parsing error with `Nothing to repeat`. `\s{0,2}` would be a safer auto-fix here I think, allowing for 0-2 of the previous token. The `{0,2}` fix is done as expected with `/a?a?/`, so it seems `\s` is being handled differently.

Further investigation shows that this looks to be an issue with the `regexp-tree` module:
```js
const {optimize} = require('regexp-tree');
const result = optimize(/\s?\s?/);
/*
_ast: {
    type: 'RegExp',
    body: { type: 'Alternative', expressions: [Array] },
    flags: ''
  },
  _source: null,
  _string: '/\\s?{2}/',
  _regexp: null,
  _extra: null
*/
```
Opened an issue on that repo: https://github.com/DmitrySoshnikov/regexp-tree/issues/216

### PR DESCRIPTION
**Pull request by Cherry**
Fixes #994

### CODE REVIEW

**Review by fisker — COMMENTED** [review]
Can you add test from that issue?

**Review by fisker — APPROVED** [review]
Thanks!

**Comment by Cherry** [comment]
Added 2 test cases, one for `/s?/s?` which shouldn't be optimised (as per https://github.com/DmitrySoshnikov/regexp-tree/issues/216#issuecomment-762073297). And then another for `/s?/s?/s?` which should.

File: test/better-regex.js [hunk 1]
```diff
@@ -70,7 +70,10 @@ test({
 		'new RegExp()',
 
 		// #472
-		'/[ ;-]/g'
+		'/[ ;-]/g',
+
+		// 994
```
Comment (fisker) [inline test/better-regex.js:75]:
```suggestion
// #994
```

**Review by fisker — APPROVED** [review]

**Comment by fisker** [comment]
Can we  also  test  `/\s{0,2}/` :smile:

**Comment by Cherry** [comment]
Added. `/\s?\s?/` and `/\s{0,2}/` are functionally equivalent and both do not get optimised - the transform comes into effect once you hit 3 optional repeating characters now from what I can tell, and as per [this test](https://github.com/DmitrySoshnikov/regexp-tree/blob/0421d57324defe30ff15a5dbb5af5c08f751a795/src/optimizer/__tests__/optimizer-integration-test.js#L119). I included a link to the comment about this.

**Comment by fisker** [comment]
Good job.

**Pull request merged**